packages/dds/cell/src/cell.ts

```diff
@@ -229,6 +231,25 @@ export class SharedCell<T = any> extends SharedObject<ISharedCellEvents<T>>
         return tree;
     }
 
+    /**
+     * {@inheritDoc @fluidframework/shared-object-base#SharedObject.captureSummaryStateCore}
+     */
+     protected captureSummaryStateCore(serializer: IFluidSerializer, fullTree: boolean): any {
```

Comment:
it would be great not to use 'any'

Reply:
I'd also prefer naming to reflect future possible usage / scenario expansion. I.e. this function captured revision, or state of DDS at given point in time. What can be done with revision / snapshot - will change over time, I do not think producing summary will be the only thing we can do with it. So, I'd rather change name not to have anything to do with summary here. 

This forces an interesting question - should fullTree be an argument to this function, or the method on returned object that produces summary. I think it should be the later, but worth exploring what it means in praxtice.

Reply:
I have same feeling for serializer arg. The intention of this function is to be fast and do minimal processing for something like property DDS that is based on immutable data. So I think anything that has relationship to summary / long processing should be in second method / method of object that this call should return.

This changes flow substantially. So I'm not sure about all the side-effects of this direction, but I'd try to see where it leasds.

Note that we can always split SharedObject into two layer - one base that has no assumptions and supports all flows.
And then derived abstract class (to be used by all DDSs other than PropertyDDS) that would hide these complexities (in the same way as SharedObject does today) and just expose single synchronous method to summarize exactly how it works today, creating serializer in first call (and keeping it around), and exposing summary and gc info at a later stage.

Reply:
I agree with getting rid of `serializer` here. It does mean bigger changes to all the DDS implementations though, since I can't just call existing snapshotCore implementations.

Reply:
You can avoid a lot of changes - I believe I provided feedback earlier in some of the comments, but having two base layers here - one that makes all these changes and allows full flexibility, and then abstract implementation of that protocol that requires existing synchronous summarization process for all DDSs other than those that want more control.
Happy to chat about it, but I think creating more structure here (i.e. returning an object) is a must

Reply:
Something like this commit? [https://github.com/microsoft/FluidFramework/pull/8422/commits/31462ac7f56fced0e9bf47642143dc9702951cce](url)

The diff is not very helpful. Basically, SharedObjectBase has:
```typescript
interface ISharedObjectSnaphost {
    summarize(serializer: IFluidSerializer, fullTree?: boolean): Promise<ISummaryTreeWithStats>;
}
class SharedObjectBase {
...
    public captureState(): ISnaphost {
        const state = this.captureStateCore();
        return { summarize: async (fullTree: boolean = false) => {
            return state.summarize(this._serializer, fullTree);
        }};
    }
    protected abstract captureStateCore(): ISharedObjectSnaphost;
}
```
and SharedObject only has:
```typescript
class SharedObject extends SharedObjectBase {
    protected captureStateCore(): ISharedObjectSnaphost {
        const snapshot = this.snapshotCore(this.serializer);
        return { summarize: async (serializer: IFluidSerializer, fullTree: boolean = false) => {
            return convertToSummaryTreeWithStats(snapshot, fullTree);
        }};
    }
    protected abstract snapshotCore(serializer: IFluidSerializer): ITree;
}
```
With fullTree moved to the async method, it wouldn't be possible to have [renamed] snapshotCore return ISummaryTreeWithStats (#8013).

Reply:
Yep, something like that